target/target.go

```diff
@@ -81,6 +83,22 @@ func (t *Target) CreateGNMIClient(ctx context.Context, opts ...grpc.DialOption)
 		go func(addr string) {
 			timeoutCtx, cancel := context.WithTimeout(ctx, t.Config.Timeout)
 			defer cancel()
+			if t.Config.SocksAddress != "" {
+				dialer, err := proxy.SOCKS5("tcp", t.Config.SocksAddress, nil,
+					&net.Dialer{
+						Timeout:   t.Config.Timeout,
+						KeepAlive: t.Config.Timeout,
+					},
+				)
+				if err != nil {
+					errC <- fmt.Errorf("%s: %v", addr, err)
+					return
+				}
```

Comment:
I would put this dialer creation as part of the `grpc.WithContextDialer` function.